### PR TITLE
Fe dec19 vitalii b

### PR DESCRIPTION
**Pull request by VitaliyBondarenko1982**
[DEMO LINK](https://VitaliyBondarenko1982.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,8 +1,67 @@
...
+import { GoodsList } from './components/GoodsList/GoodsList';
+import { Button } from './components/Button/Button';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const URL = 'https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json';
+
+const getGoodsFromServer = () => {
+  const getGoods = fetch(URL)
```
Comment (YuriiHoliuk) [inline src/App.tsx]: `getGoods` - the name of the function but `getGoods` is not a function

File: src/App.tsx [hunk 2]
```diff
@@ -1,8 +1,67 @@
...
 import './App.css';
+import { GoodsList } from './components/GoodsList/GoodsList';
+import { Button } from './components/Button/Button';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const URL = 'https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json';
+
+const getGoodsFromServer = () => {
```
Comment (YuriiHoliuk) [inline src/App.tsx]: just getGoods

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React, { FC } from 'react';
+
+type Props = {
```
Comment (YuriiHoliuk) [inline src/components/Button/Button.tsx]: use interface instead of type when possible

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  clickHandler: () => void;
```
Comment (YuriiHoliuk) [inline src/components/Button/Button.tsx]: не придумуй велосипед, назви onClick

File: src/App.tsx [hunk 5]
```diff
@@ -1,8 +1,67 @@
...
+  const getGoods = fetch(URL)
+    .then(response => response.json());
+
+  return getGoods;
+};
+
+interface State {
+  goods: {id: number; name: string; color: string}[];
```
Comment (YuriiHoliuk) [inline src/App.tsx]: create good interface

File: src/App.tsx [hunk 6]
```diff
@@ -1,8 +1,67 @@
...
+        goods: data,
+      }));
+  };
+
+  showFiveGoods = () => {
+    const getGoods = getGoodsFromServer();
+
+    getGoods
+      .then(data => this.setState({
+        goods: data.sort((a: {name: string}, b: {name: string}) => a.name.localeCompare(b.name))
```
Comment (YuriiHoliuk) [inline src/App.tsx]: create good interface

File: src/App.tsx [hunk 7]
```diff
@@ -1,8 +1,67 @@
...
+          .slice(0, 5),
+      }));
+  };
+
+  showRedGoods = () => {
+    const getGoods = getGoodsFromServer();
+
+    getGoods
+      .then(data => this.setState({
+        goods: data.filter((good: { color: string }) => good.color === 'red'),
```
Comment (YuriiHoliuk) [inline src/App.tsx]: create good interface

File: src/components/GoodsList/GoodsList.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React, { FC } from 'react';
+
+type Props = {
+  goods: {id: number; name: string; color: string}[];
```
Comment (YuriiHoliuk) [inline src/components/GoodsList/GoodsList.tsx]: create good interface

File: src/App.tsx [hunk 9]
```diff
@@ -1,8 +1,67 @@
...
-  <h1>Dynamic list of Goods</h1>
-);
+const URL = 'https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json';
+
+const getGoodsFromServer = () => {
+  const getGoods = fetch(URL)
+    .then(response => response.json());
+
+  return getGoods;
+};
```
Comment (YuriiHoliuk) [inline src/App.tsx]: not required but will be great to move these stuff to separate file/files

File: src/App.tsx (hunk 1 above)
Comment (VitaliyBondarenko1982) [reply src/App.tsx]: ОК

File: src/App.tsx (hunk 2 above)
Comment (VitaliyBondarenko1982) [reply src/App.tsx]: ОК

File: src/components/Button/Button.tsx (hunk 3 above)
Comment (VitaliyBondarenko1982) [reply src/components/Button/Button.tsx]: ОК

File: src/components/Button/Button.tsx (hunk 4 above)
Comment (VitaliyBondarenko1982) [reply src/components/Button/Button.tsx]: ОК

File: src/App.tsx (hunk 5 above)
Comment (VitaliyBondarenko1982) [reply src/App.tsx]: ОК

File: src/App.tsx (hunk 6 above)
Comment (VitaliyBondarenko1982) [reply src/App.tsx]: ОК

File: src/App.tsx (hunk 7 above)
Comment (VitaliyBondarenko1982) [reply src/App.tsx]: ОК

File: src/components/GoodsList/GoodsList.tsx (hunk 8 above)
Comment (VitaliyBondarenko1982) [reply src/components/GoodsList/GoodsList.tsx]: ОК

File: src/App.tsx (hunk 9 above)
Comment (VitaliyBondarenko1982) [reply src/App.tsx]: ОК

File: src/components/GoodsList/GoodsList.tsx [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+type Props = {
+  goods: Good[];
+};
+
+export interface Good {
+  id: number;
+  name: string;
+  color: string;
+}
```
Comment (YuriiHoliuk) [inline src/components/GoodsList/GoodsList.tsx]: better to move to separate types constants file or to the global types

**Pull request closed without merging**